### ISSUE TITLE
Would be nice if all UI elements were accessible without mouse

### ISSUE THREAD

**Issue by gulkily** [sqlitebrowser/sqlitebrowser#2215]
Describe the new feature
--------------------------
Feature description: All of the UI elements, but especially the most essential ones, accessible with keyboard only, either via Tab-based navigation, or accelerator keys, or shortcut keys.

For example, on startup, I am able to navigate the tabs with Ctrl+Tab. This is great. And I can select the table in the dropdown by first pressing the Alt+T accelerator. Good, so far.

However, when I get to the tab where I can execute SQL I type, I cannot (that I've figured out how to) switch to the tab where I enter the code. I must resort to clicking using the mouse cursor.

Additional info
---------------
Please answer these questions before submitting your feature request.

#### Is your feature request related to an issue? Please include the issue number.

No

#### Does this feature exist in another product or project? Please provide a link.

Yes, most applications support this.

#### Do you have a screenshot? Please add screenshots to help explain your idea.

N/A

**Comment by yamamov** [comment]
I also want to use the keyboard to access the filters by column in the browser tab. It's best to have a shortcut key, but it doesn't matter if the focus is just shifted by using tab keys, etc.

**Comment by justinclift** [comment]
Yeah, this is a good idea. :smile:

**Comment by horst-p-w-neubauer** [comment]
Would it be possible to have Ctrl+Shift+T to toggle the SQL-Tabs and set the Cursor into the SQL-Editor? As Ctrl+T opens a new Tab Ctrl+Shift+T feels to be intuitive for me.

**Comment by horst-p-w-neubauer** [comment]
Is there a shortcut to switch between SQL-editor and result pane? I think, that would be nice too.

**Comment by mgrojo** [comment]
Doesn't Alt+4 work for you? In 3.12 alpha, 4 must be exactly the position of the tab.

Ok, if the editor isn't previously focused in that tab, then you have to use the Tab key to switch to it, but that is normal behaviour, isn't it?

> Is there a shortcut to switch between SQL-editor and result pane? I think, that would be nice too.
Should be the tab key, but that it's caught by the editor for inserting the tab character. Yes, it should be a way out. I haven't found any key already-working key combination, so problaby we should implement one.

**Comment by horst-p-w-neubauer** [comment]
@mgrojo: Alt+4 Enters activates the most right of the top-most tabs ("Execute SQL") - my suggestion was intended for the SQL-Edit Tabs inside "Execute SQL"
![grafik](https://user-images.githubusercontent.com/28151942/80309307-8f95d800-87d4-11ea-97d0-ce56f839320f.png)

**Comment by mgrojo** [comment]
> @mgrojo: Alt+4 Enters activates the most right of the top-most tabs ("Execute SQL") - my suggestion was intended for the SQL-Edit Tabs inside "Execute SQL"

I don't reproduce that. I open a database file, press Alt+4, and the focus is in the blank editor tab. Of course, if you move the focus in the Execute SQL tab, and then you come back, the focus is not in the editor, but you can press Tab or Shift+Tab to return to it. I'm not in favor of adding new keyboard shortcuts only to focus certain widgets; the general mechanism is to navigate using tab. Maybe we could assume that pressing Alt+4, apart from changing to the Execute SQL, it should change focus to the editor, since it is what the user probably want. I'd prefer that, in order to save a new shortcut, but even then, I don't know if we are going to get in the way of some users' customs.

**Comment by horst-p-w-neubauer** [comment]
I just found, that Ctrl+Tab and Ctrl+Shift+Tab allready toggles between Editor Tabs in Execute SQL, so no need for Ctrl+Shift+T. 
But there remain two challenges:
# I cant find a solution to switch between editEditor and tableResult in the sqlExecutionArea by keyboard. I often missed this. 
# There is an inconsistency. As said Ctrl+Tab  toggles thru the editor tab - this is only when your cursor is in the editEditor. When the cursor is in tableResult Ctrl+Tab does some useles (IMHO) cell movements instead. In my opinion the Tab (e.g. "SQL 2") is associated to the complete sqlExecutionArea - editEditor as well as tableResult. Key behavior that adresses Elements outside sqlExecutionArea (Ctrl+Tab) should be the same from everywhere inside sqlExecutionArea - regadles if you are in editEditor or in tableResult. Having two editor-tabs and your cursor in tableResult in "SQL 1" and in editEditor in "SQL 2" and "SQL 2" active. Pressing Ctrl+Tab switches to "SQL 1" - but into tableResult. Pressing Ctrl+Tab again moves the cell selection - that is confusing.

**Comment by mgrojo** [comment]
Thanks for the analysis, @horst-p-w-neubauer. I've made two changes to address the issues that you have mentioned:
# It's possible to focus out of the SQL Editor using Ctrl+PgDown (it wasn't in use by QScintilla, our editor library)
# Ctrl+Tab and Ctrl+Shift+Tab works in all the widgets in the SQL Execution area and
provides the same functionality that the one provided by default by Qt
(switch for- and backwards through the tabs).
@horst-p-w-neubauer and @gulkily, could you test the [tomorrow's nightly build](https://nightlies.sqlitebrowser.org/) and report any undesired side effect or pending issue?

**Comment by justinclift** [comment]
Should this go into 3.12 too?

**Comment by mgrojo** [comment]
> Should this go into 3.12 too?

Yes, it could be, provided that people confirm that it works well and doesn't cause any regression.

**Comment by horst-p-w-neubauer** [comment]
@mgrojo , Ctrl+PgDown works fine to get out of the SQL Editor itself. But there the trouble starting again. 
1. If you have results you get stuck in the result table. No Ctrl+PgDown gets you out of this.  (Well - not exactely. You can switch to the next editor tab (Ctrl+Tab) and back (Ctrl+Shift+Tab) and find the focus on the SQL Editor. But I see this just as a workarround).
2. If you don't have any results the focus changes to - well, I don't know.  Maybe to the empty result-pane - which should not be able to take focus IMHO.
3. My intuition would be to rollback my last Ctrl+PgDown command with a following Ctrl+PgUp to get back to the SQL Editor. That's not working.

**Comment by horst-p-w-neubauer** [comment]
The consistent switching of editor-Tabs inside the SQL execution area is realy nice!

**Comment by horst-p-w-neubauer** [comment]
All in all I'd think we should have 
1. a consistent way to reach the result pane - if there are any results.
2. a consistent way to go back to the SQL Editor pane
3. better focus indication all over the gui
4. a consistent way to get from the focused Tab-Header (e.g. "SQL 4") into the SQL-Editor with just one key-stroke - I'd suggest: Return

**Comment by mgrojo** [comment]
>     3\. My intuition would be to rollback my last Ctrl+PgDown command with a following Ctrl+PgUp to get back to the SQL Editor. That's not working.

@horst-p-w-neubauer, you are right about that. It's already pushed.

> All in all I'd think we should have
> 
>     1. a consistent way to reach the result pane - if there are any results.
> 
>     2. a consistent way to go back to the SQL Editor pane

Ctrl+PgDown and Ctrl+PgUp cover now that, don't they?
> 
>     3. better focus indication all over the gui

I don't understand. Do you mean that there are widgets, which don't indicate in some way that they have focus? That is the toolkit's mission and it's very dependent on the style.

> 
>     4. a consistent way to get from the focused Tab-Header (e.g. "SQL 4") into the SQL-Editor with just one key-stroke - I'd suggest: Return

That might be platform-dependent. Apparently, my tab-header never gets focus. Do you have steps to reproduce it? In that case, where is <kbd>Tab</kbd> transferring the focus?

**Comment by gulkily** [comment]
It's not clear to me how to install the nightly build on Fedora, so I'm unable to test at this time.

I realized I should have attached clear steps for testing, so here they are:

1. Open by running, e.g. "sqlitebrowser example.sqlite3"
2. Press Ctrl+Tab to switch to the Execute tab (this already works)
3. Press Tab to switch to the SQL entry field.

Expected: SQL entry field gets focus on the firt press of Tab key

Actual: Nothing actually seems to get focus

I'll paste these into the issue too.

**Comment by horst-p-w-neubauer** [comment]
@mgrojo Ctrl+PgUp doesn't work for me - regardless if there is a result table or not. I'll make some screenshots in the next days.

**Comment by justinclift** [comment]
@gulkily For Fedora, it needs to be manually compiled.  Fortunately, there are decent well travelled instructions here (pretty much "cut-n-paste"):

* https://github.com/sqlitebrowser/sqlitebrowser/blob/master/BUILDING.md#centos--fedora-linux

Is that your kind of thing? :smile:

**Comment by mgrojo** [comment]
> @mgrojo Ctrl+PgUp doesn't work for me - regardless if there is a result table or not. I'll make some screenshots in the next days.

@horst-p-w-neubauer maybe you don't see any feedback of the focus, but it's already there? Try to press tab or the cursor keys to move the current cell. Maybe it's only that, otherwise, I don't know where the problem could lie.
